spec/rails_translation_manager/locale_checker/missing_declared_locales_spec.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe MissingDeclaredLocales do
+  context "when there are missing locales" do
+    let(:all_locales) do
+      [
+        {
+          locale: :en,
+          keys: %w[browse.same_key]
+        },
+        {
+          locale: :cy,
+          keys: %w[browse.same_key]
+        },
+        {
+          locale: :pt,
+          keys: %w[browse.same_key]
+        }
+      ]
+    end
+
+    let(:actual_locales) do
+      [:en, :cy]
+    end
+
+    before do
+      I18n.available_locales << %i[pt]
+    end
+
+    it "outputs the missing locales" do
+      expect(described_class.new(all_locales).report)
+        .to eq(
+          <<~OUTPUT.chomp
+          \e[31m[ERROR]\e[0m No locale files detected for:
+
+          pt
+
+          \e[1mEither create these locale files or remove these locales from your I18n `available_locales` config.\e[22m
+          OUTPUT
+        )
+    end
+  end
+
+  context "when there aren't missing locales" do
+    let(:all_locales) do
+      [
+        {
+          locale: :en,
+          keys: %w[browse.same_key]
+        },
+        {
+          locale: :cy,
+          keys: %w[browse.same_key]
+        }
+      ]
+    end
+
```

Comment:
As with the other test, I think it would be a good idea to make it explicit that `en`/`cy` exist, and not rely on the values set in the spec_helper.rb. (I think in a future refactor we'll want to get rid of that, tbh!)